### PR TITLE
TAJO-1317: Parallel Test Executions on Tajo Core Project

### PR DESCRIPTION
**Pull request by ykrips**

### CODE REVIEW

**Comment by ykrips** [comment]
It should be separated compilation and test phase on maven build, because parallel compilation is not supported by maven.

**Comment by ykrips** [comment]
This build failure already reported to MNG jira. It needs to find a workaround for this build issue. More details can be found in below link.
https://jira.codehaus.org/browse/MNG-5750

File: tajo-core/pom.xml [hunk 1]
```diff
@@ -690,6 +696,37 @@
...
+    <profile>
+      <id>parallel-test</id>
+      <activation>
+        <activeByDefault>false</activeByDefault>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
```
Comment (jinossy) [inline tajo-core/pom.xml]: Could you add latest version? In my test, I met the ClassNotFoundExeception

Comment (ykrips) [inline tajo-core/pom.xml]: Thank you for reporting a issue. I did not test on MacOS, and found that maven on MacOS searches the invalid classname like 'rg.apache...'. I will keep finding a workaround on this issue.

Comment (jinossy) [inline tajo-core/pom.xml]: Sorry for the confusion. I was test successfully with TAJO-1322, TAJO-1323 on surefire-2.18.1

Comment (ykrips) [inline tajo-core/pom.xml]: Thank you for posting test results. Finally found out that reusing test processes cause the classnotfoundexception.

Comment (jinossy) [inline tajo-core/pom.xml]: @ykrips 
If reuseForks is false, integration test is too many restarted the testing cluster
I've successfully ran the parallel execution with enabled reuseForks
```
mvn clean install -Pparallel-test,hcatalog-0.12.0 -DLOG_LEVEL=WARN -Dmaven.fork.count=3
```

Comment (ykrips) [inline tajo-core/pom.xml]: @jinossy ,
Thank you for reporting the result. I will try again with reuseForks option.

**Comment by jinossy** [comment]
@ykrips 
It seems like surefire plugin can’t handle the ServiceLoader in CompressionCodecFactory
Can you test again? useSystemClassLoader=true

**Comment by ykrips** [comment]
Just committed. And also verified on my ubuntu box.

**Comment by jinossy** [comment]
+1 LGTM.
Ship it!

**Pull request closed without merging**